File: Problems/Python/Medium/Array/1833. Maximum Ice Cream Bars.py
/*
Runtime: 1000 ms, faster than 60.97% of Python3 online submissions for Maximum Ice Cream Bars.
Memory Usage: 28 MB, less than 53.28% of Python3 online submissions for Maximum Ice Cream Bars.
*/
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        ans = 0
        costs.sort()
        for i in range(0,len(costs)):
            if coins >= costs[i]:
                coins-=costs[i]
                ans+=1
            else:
                break
        return ans

File: Problems/C++/Medium/Array/1833. Maximum Ice Cream Bars.cpp
/*
Runtime: 144 ms, faster than 99.56% of C++ online submissions for Maximum Ice Cream Bars.
Memory Usage: 76.6 MB, less than 26.95% of C++ online submissions for Maximum Ice Cream Bars.
*/
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int ans = 0;
        for(int i = 0; i<costs.size(); i++){
            if(costs[i]<=coins){
                ans++;
                coins-=costs[i];
            }else{
                break;
            }
        }
        return ans;
    }
};